Oereblex 1.2.0 as default and support notice document (HintRecord)

ger-benjamin opened the pull request:
For GEO-2697 and https://github.com/openoereb/pyramid_oereb/issues/959

 - Default oereblex version is now 1.2.0.
 - Notice document will be "classed" as "HintRecord" (Hinweis).

For the point: 
```
Furthermore, it is requested from the canton AI, that it's sufficient for an geometr
to then have either a geolink which refers to the "Gesetzliche Grundlagen" or
a link to the "Weitere Informationen und Hinweise".
```
It should already work with the `reduced` "mode" (flavour). But not with others as the official requirement from the specification spcify this point: "Eigentumsbeschränkung ohne Rechtsvorschrift ist nur bei einem reduzierten Auszug zulässig".

Can someone with a pyramid_oereb - oereblex application take this commit and test ? (that would be quicker than make a test environment with oereblex)

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -14,6 +14,13 @@ Version 1.7.2 (CURRENT DEV)
 Changes thus far are relevant only for users of MapFish Print.
 
+Oereblex related changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
It needs to be exactly the same length as the title, otherwise doc & package generation will fail

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -14,6 +14,13 @@ Version 1.7.2 (CURRENT DEV)
 This is the current development version, not yet released.
 Changes thus far are relevant only for users of MapFish Print.
```
this sentence can now be removed

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -14,6 +14,13 @@ Version 1.7.2 (CURRENT DEV)
 
+Oereblex related changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Pyramid Oereb now support and use by default geoLink schema version 1.2.0:
```
supports and uses

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -14,6 +14,13 @@ Version 1.7.2 (CURRENT DEV)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Pyramid Oereb now support and use by default geoLink schema version 1.2.0:
+
+ * New doctype 'notice' (will be classed as 'HintRecord'). 'Notice' can have no authority nor enactment_date
```
classed -> classified

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -14,6 +14,13 @@ Version 1.7.2 (CURRENT DEV)
+Pyramid Oereb now support and use by default geoLink schema version 1.2.0:
+
+ * New doctype 'notice' (will be classed as 'HintRecord'). 'Notice' can have no authority nor enactment_date
+ * New document attribute 'language' and file attribute 'description' are currently not used by pyramid_oereb.
```
should we add here something like: "but are now available to custom code, for example for document title generation."

(several cantons have custom code regarding this)

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

jwkaltz commented:
> The code looks good to me. Unfortunately, I cannot test the code on our installation as there is still some migration effort necessary to update to the latest version of pyramid_oereb. We still use a couple of customized sources and readers from earlier versions which we have planned to eliminate. Furthermore, we currently do not have any notices in our legal data in OEREBlex.

Thanks for your code review @kdeininger . We have asked Max from Geoinfo to test it, with a test release of pyramid_oereb.

ger-benjamin commented:
Thanks for the review @kdeininger  @jwkaltz 

After some test with @MaxHoeferGEOINFO , we have seen some issues (corrected in my last commit):
 - I've added a flag `related_notice_as_main` to have notice directly as main document.
- pyramid_oereb expect an `enactment_date` for documents. Oereblex allows to have none. Max will ask to the oereblex team to have an enactment date in notices. But for now I add a `01.0.1.1979` enactment_date (for notice only).
 - And same for the `authority` in notice. I've added `-` as default value for notice documents. 

I'll do a 1.7.2.dev2 and test again with @MaxHoeferGEOINFO

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

ger-benjamin commented:
With my last commit, I remove the default value for `authority_as_web` and add support of empty `authority_as_web` in pyramid_oereb_mfp (`pyramid_oereb` already support having no `authority_as_web` (`office_at_web`)).
See: https://github.com/openoereb/pyramid_oereb_mfp/pull/62

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)

jwkaltz commented on doc/source/changes.rst:
````diff
@@ -12,14 +12,29 @@ to adapt in your project configuration, database etc. when upgrading to a new ve
 MapFish Print should set the print configuration parameter ```with_geometry``` to ```False```
 to improve performance.
 
+It's now possible to print report with missing OfficeAtWeb information in documents.
````
Small change suggestion:

It is now possible to print reports with missing OfficeAtWeb information in documents (OfficeAtWeb is an optional attribute in the specification).

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -12,14 +12,29 @@ to adapt in your project configuration, database etc. when upgrading to a new ve
+Pyramid Oereb now supports and uses by default geoLink schema version 1.2.0:
+
+ * New doctype 'notice' (will be classified as 'HintRecord'). If you want to add related notice as
+   additional legal provision directly to the public law restriction, You can set the new oereblex
```
You -> you

jwkaltz commented on doc/source/changes.rst:
```diff
@@ -12,14 +12,29 @@ to adapt in your project configuration, database etc. when upgrading to a new ve
+   'related_notice_as_main' flag in the config of the project.
+ * 'Notice' can have no authority nor enactment_date. In this case, the enactment date will be
+   '01.01.1970' and the authority '-'.
+ * 'Notice' can have no authority_at_web. That was not supported by MapFish Print. If you use MapFish Print
```
Please move this to the "MapFish Print related changes" section

ger-benjamin replied:
MFP section as already a note (without oereblex mention).
This mention is for user using oereblex (with MFP). So I let it there.

pfirpfel commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/contrib/sources/document.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5067020&bid=16672885)
